docs/use-cases/youtube_studyplan_agent.md

```diff
@@ -0,0 +1,163 @@
+
+# 🎓 Optimization Techniques Study Plan Generator
+
+**Overview**
+
+This example uses OpenAGI to generate a detailed study plan for optimization techniques. It leverages the Gemini model and various OpenAGI components to create a study plan with YouTube video references for each topic, neatly written into a file in markdown format.
+
+## 📦 Required Libraries
+
+First, import the required libraries. These libraries enable the agents to plan tasks, write and read files, search YouTube for videos, and leverage the Gemini large language model (LLM).
+
+```python
+from openagi.agent import Admin
+from openagi.worker import Worker
+from openagi.actions.files import WriteFileAction, ReadFileAction
+from openagi.actions.tools.youtubesearch import YouTubeSearchTool
+from openagi.llms.gemini import GeminiModel
+from openagi.memory import Memory
+from openagi.planner.task_decomposer import TaskPlanner
+import os
+```
+
+## ⚙️ Setup Environment
+
+Configure the necessary environment variables. These include your API key and Gemini model details, which are essential for the LLM to generate relevant outputs.
+
+```python
+# Define the llm
+os.environ['GOOGLE_API_KEY'] = "<your-api-key>"
+os.environ['Gemini_MODEL'] = "gemini-1.5-flash"  # You can use any available model
+os.environ['Gemini_TEMP'] = "0.5"
+
+# Load the LLM configuration from environment
+config = GeminiModel.load_from_env_config()
+llm = GeminiModel(config=config)
+```
+
+## 🗂️ Task Planning
+
+A task planner is defined to help in decomposing tasks for creating a study plan. This planner uses the Gemini LLM and has the option to allow human intervention during task execution.
+
+```python
+# Define the planner
+planner = TaskPlanner(
+    human_intervene=True,  # Set to False for full automation
+    llm=llm,
+    retry_threshold=5
+)
+```
+
+## 🧑‍💼 Admin and Workers
+
+Set up the Admin agent, which manages the workflow and assigns tasks to worker agents. The admin ensures that the study plan is output in markdown format.
+
+```python
+# Define the admin
+admin = Admin(
+    llm=llm,
+    planner=planner,
+    max_steps=5,
```

Comment:
Admin have no max_steps=5, replace with max_iterations

Reply:
done, but i think we have to change this on docs too:
![image](https://github.com/user-attachments/assets/da6d3e92-f179-4ef5-b450-f2e4c4cc7b8f)